.travis.yml

```diff
@@ -1,17 +1,60 @@
-language: cpp
 dist: trusty
 sudo: required
 
-before_install:
-  - . .travis.setup
+env:
+  - CUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-8.0
 
-before_script:
-  - mkdir build
-  - cd build
-  - cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-8.0 ..
+_install: &_install
+  - . .travis.install
+  - pip install numpy auditwheel==1.5 setuptools twine
+  - python3 setup.py bdist_wheel
+
+_deploy: &_deploy
+  provider: script
+  script:
+    - auditwheel repair -w dist dist/*
+    - twine upload dist/*manylinux* -u $PYPI_LOGIN -p $PYPI_PASS
+  skip_cleanup: true
+  on:
+    tags: true
+
+matrix:
+  include:
+    - language: cpp
+      install:
+        - . .travis.install
+        - mkdir build
+        - cd build
+        - cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCUDA_TOOLKIT_ROOT_DIR=$CUDA_TOOLKIT_ROOT_DIR ..
+        - make -j2
+      script: skip
+    - language: python
+      python: 3.4
+      install: *_install
+      script: skip
+      deploy:
+        provider: script
+        script:
+          - auditwheel repair -w dist dist/*
+          - twine upload dist/*manylinux* -u $PYPI_LOGIN -p $PYPI_PASS
+        skip_cleanup: true
+        on:
+          tags: true
+    - language: python
+      python: 3.5
+      before_install:
```

Comment:
You don't need virtualenv in this ephemeral environment, nothing to isolate from. Each job runs in a separate instance.